interpreter/cling/lib/UserInterface/UserInterface.cpp

```diff
@@ -66,6 +67,43 @@ namespace {
 
     textinput::TextInput* operator -> () { return &m_Input; }
   };
+
+  llvm::SmallString<512> GetHistoryFilePath() {
+    if (getenv("CLING_NOHISTORY")) {
+      return {};
+    }
+
+    if (const char* HistFileEnvvar{std::getenv("CLING_HISTFILE")}) {
+      return llvm::StringRef{HistFileEnvvar};
+    }
+
+    // According to XDG Base Directory Specification, the history file
+    // locations and names, ordered in priority, are:
+    //
+    // ${XDG_STATE_HOME}/cling/history
+    // ~/.local/state/cling/history
+    // ~/.cling_history
+    const char* StateHome{std::getenv("XDG_STATE_HOME")};
+
+    if (!StateHome) {
+      StateHome = "~/.local/state";
+    }
+
+    llvm::SmallString<512> FilePath;
+
+    if (!llvm::sys::fs::real_path(StateHome, FilePath, true) &&
```

Comment:
I would probably separate the path resolution, i.e. `real_path()`, from the rest of the `if` statement.

Reply:
Done.